apps/frontend/src/utilities/delta_util.ts

```diff
@@ -163,6 +165,68 @@ export function get_eval_start_time(
   return null;
 }
 
+export function getResultsSetExecutionTime(
+  resultsSet: SourcedContextualizedEvaluation
+): number {
+  let time = 0;
+  resultsSet.contains.forEach((value) => {
+    value.contains.forEach((control) => {
+      control.hdf.segments?.forEach((segment) => {
+        time += segment.run_time || 0;
+      });
+    });
+  });
+  return time;
+}
+
+export function getControlCount(
+  resultsSet: SourcedContextualizedEvaluation
+): number {
+  let count = 0;
+  resultsSet.contains.forEach((value) => {
+    count += value.contains.length;
+  });
+  return count;
+}
+
+export function getCompletedControlCount(
+  resultsSet: SourcedContextualizedEvaluation
+): number {
+  let count = 0;
+  resultsSet.contains.forEach((value) => {
+    value.contains.forEach((control) => {
+      if (
+        ![
+          'Not Applicable',
+          'Not Reviewed',
+          'Profile Error',
+          'From Profile'
+        ].includes(control.hdf.status)
+      ) {
+        count += 1;
+      }
+    });
+  });
+  return count;
+}
+
+export function calculateCompliance(
+  resultsSet: SourcedContextualizedEvaluation
+): number {
+  const filter: Filter = {fromFile: [resultsSet.from_file.uniqueId]};
+  const passed = StatusCountModule.countOf(filter, 'Passed');
+  const total =
+    passed +
+    StatusCountModule.countOf(filter, 'Failed') +
+    StatusCountModule.countOf(filter, 'Profile Error') +
+    StatusCountModule.countOf(filter, 'Not Reviewed');
+  if (total === 0) {
+    return 0;
+  } else {
+    return Math.round((100.0 * passed) / total);
+  }
```

Comment:
This function is identical to the one used to calculate compliance elsewhere. Can you just add a `StatusCountModule.calculateCompliance` method and then reuse it everywhere?